### PR TITLE
[jsk_apc2016_common] add cpp message synchronizer

### PR DESCRIPTION
**Pull request by yuyu2172**
I wrote a cpp code that synchronizes topics and republish a single topics.

This solves issue raised in #1396. Apparently, the problem persisted through debugging because Python is too slow to process topics with large size.

### CODE REVIEW

File: jsk_apc2016_common/src/rbo_topic_synchronizer.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#include <message_filters/subscriber.h>
+#include <message_filters/time_synchronizer.h>
+#include <sensor_msgs/Image.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <message_filters/sync_policies/approximate_time.h>
```
Comment (knorth55) [inline jsk_apc2016_common/src/rbo_topic_synchronizer.cpp]:
```
<build_depend>sensor_msgs</build_depend>
<build_depend>message_filters</build_depend>

<run_depend>sensor_msgs</run_depend>
<run_depend>message_filters</run_depend>
```
is required in `package.xml`, i think

Comment (yuyu2172) [inline jsk_apc2016_common/src/rbo_topic_synchronizer.cpp]: oh thanks!

**Comment by wkentaro** [comment]
lgtm

**Comment by wkentaro** [comment]
but this should be merged in master as I said in slack.

**Comment by knorth55** [comment]
thanks. i follow your decision

**Comment by yuyu2172** [comment]
I will reopen PR to merge this to master

**Pull request closed without merging**